core/server/api/v2/subscribers.js

```diff
@@ -0,0 +1,202 @@
+const fs = require('fs-extra');
+const models = require('../../models');
+const fsLib = require('../../lib/fs');
+const common = require('../../lib/common');
+
+const subscribers = {
+    docName: 'subscribers',
+    browse: {
+        options: [
+            'page',
+            'limit',
+            'fields',
+            'filter',
+            'order',
+            'debug'
+        ],
+        permissions: true,
+        query(frame) {
+            return models.Subscriber.findPage(frame.options);
+        }
+    },
+
+    read: {
+        headers: {},
+        data: [
+            'id',
+            'email'
+        ],
+        permissions: true,
+        query(frame) {
+            return models.Subscriber.findOne(frame.data);
+        }
+    },
+
+    add: {
+        statusCode: 201,
+        headers: {},
+        permissions: true,
+        query(frame) {
+            return models.Subscriber.getByEmail(frame.data.subscribers[0].email)
```

Comment:
My PR #9988 now supports resource validation (currently in shared, so the structure would be for all api versions the same, but i don't think it will change soon). 

```
validation
  data
     email
        required: true
```

We could also change it to

```
validation
  data
     subscribers.0.email
        required: true
```

Buuut i was not sure about it. Can still change in the future, if multiple api versions have different input formats.

Reply:
I think it would be better to have serialization run before validation - that way validators can have an expected object type to work with - the raw resources. (Or at least make this configurable - which is what i'm going to do in my PR)

```js
// input serializer
module.exports = {
  add(frame) {
    frame.data = frame.data.subscribers[0];
  }
}
```

```js
// controller
module.exports = {
  docName: 'subscribers',
  add: {
    data: ['email'],
    validation: {
      data: {
          email: {required: true}
      }
    }
  }
}
```

Reply:
We had a sync call in the morning about this.
See [commit](https://github.com/TryGhost/Ghost/commit/7a73dfd9bc7b089b5c03537500a4765f86460124) in master. This was part of my PR, see #9988 and i extracted it.

A differentiation for data validation for read/browse and data validation for edit/add was not fully implemented in the frame.